src/main/clojure/clara/rules/compiler.clj

```diff
@@ -2036,6 +2036,14 @@
          (vary-meta production assoc ::rule-load-order (or n 0)))
        (range) productions))
 
+
+(defn remove-blacklisted-rules
+  "If an option :blacklist was provided by the user, we filter out the rules indicated."
+  [options rules]
+  (let [blacklist (set (map str (:blacklist options)))
+        not-contains? (complement contains?)]
+    (filter #(not-contains? blacklist (:name %)) rules)))
```

Comment:
Instead of always filtering the productions regardless of if the option was provided, could we first do a check if the option is provided then filter if it was?

Reply:
I think it would be better to check, in case we have many rules the time wasted doing nothing can be significant. 

is the following approach reasonable?
```clojure
(if (empty? blacklist) rules (filter...))
```

Reply:
Yes, i think that will work perfectly.

Reply:
Should I rename the variable `rules` to productions to follow the reasoning of the project?

Reply:
Looks good to me